core/lib/src/response/responder.rs

```diff
@@ -217,34 +219,142 @@ impl<'r> Responder<'r> for String {
     }
 }
 
+fn try_range_response<'r, B: io::Read + io::Seek + 'r>(req: Request<'r>, mut body: B) -> Result<response::Response<'r>, B> {
```

Comment:
This signature does feel really awkward, but that's mostly because of the `Seek` bound, right?

Reply:
Not really because of Seek (this implementation panics, if seeking fails), but to provide a way to recover of other errors. I could move those checks into `respond_to` instead. That would likely fix the awkwardness, but result in some code-duplication.

Reply:
I should have explained that more clearly: I found it awkward because this isn't something like `Response` -> `Response`, but instead this function requires the caller to move the body in and out. That's why I was hoping a `Responder` bounded on `Seek` would make that go away.